src/Base/BaseAttributeBag.php

```
@@ -0,0 +1,197 @@
+     *
+     * @throws InvalidArgumentException if key normalization fails.
+     *
+     * @phpstan-param mixed[] $attributes
+     * @phpstan-param scalar|Stringable|UnitEnum|Closure(): mixed $value
+     */
+    public static function set(
+        array &$attributes,
+        mixed $key,
+        bool|float|int|string|Closure|Stringable|UnitEnum|null $value,
+        string $prefix = '',
+        bool $boolToString = false,
```

PHPMD (2.15.0)

[error] The method set has a boolean flag argument $boolToString, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)